File: src/Domain/Entities/CampaniaMontanera.cs
using TrazabilidadIberica.Domain.Common;
using TrazabilidadIberica.Domain.Enums;

namespace TrazabilidadIberica.Domain.Entities;

public class CampaniaMontanera : BaseEntity
{
    public Guid FincaId { get; set; }
    public int Temporada { get; set; }
    public DateTime FechaInicio { get; set; }
    public DateTime? FechaFin { get; set; }
    public decimal HectareasUtilizadas { get; set; }
    public int CapacidadMaxAnimales { get; set; }
    public EstadoCampania EstadoCampania { get; set; }
    public string? NumAutorizacionDO { get; set; }

    public Finca Finca { get; set; } = null!;
    public ICollection<EntradaMontanera> EntradasMontanera { get; set; } = new List<EntradaMontanera>();
}

File: frontend/src/app/core/models/campania-montanera.ts
export enum EstadoCampania {
  Planificada = 0,
  Activa = 1,
  Cerrada = 2,
}

export interface CampaniaMontanera {
  id: string;
  fincaId: string;
  temporada: number;
  fechaInicio: string;
  fechaFin: string | null;
  hectareasUtilizadas: number;
  capacidadMaxAnimales: number;
  estadoCampania: EstadoCampania;
  numAutorizacionDO: string | null;
  finca?: { id: string; nombre: string };
  createdOffline: boolean;
  clientId: string | null;
  createdAt: string;
  updatedAt: string;
  deletedAt: string | null;
}
